### PR TITLE
[5.3] Single middleware pipeline

### PR DESCRIPTION
**Pull request by JosephSilber**
1. Removes a lot of code duplication.
2. Properly sorts all middleware according to priority.
3. Controller middleware now all properly fire `terminate`.

### CODE REVIEW

File: src/Illuminate/Routing/ControllerDispatcher.php [hunk 1]
```diff
@@ -18,82 +14,50 @@ class ControllerDispatcher
     protected $router;
 
     /**
-     * The IoC container instance.
-     *
-     * @var \Illuminate\Container\Container
-     */
-    protected $container;
```
Comment (crynobone) [inline src/Illuminate/Routing/ControllerDispatcher.php]: https://github.com/laravel/framework/blob/5.3/src/Illuminate/Routing/RouteDependencyResolverTrait.php#L85

`RouteDependencyResolverTrait` is using `$container` from `ControllerDispatcher`.

Comment (crynobone) [inline src/Illuminate/Routing/ControllerDispatcher.php]: screen shot 2016-06-22 at 10 11 20 pm

Comment (taylorotwell) [inline src/Illuminate/Routing/ControllerDispatcher.php]: Ping @JosephSilber

Comment (JosephSilber) [inline src/Illuminate/Routing/ControllerDispatcher.php]: Fixed here: https://github.com/laravel/framework/pull/14108

**Pull request merged**